### DISCUSSION TITLE
npm audit false positive flag

### DISCUSSION THREAD

**Discussion by chris-at-lemon** [nextauthjs/next-auth#11995] [Help] [answered]
Hi all, here's what my audit says:

```
cookie  <0.7.0
cookie accepts cookie name, path, and domain with out of bounds characters - https://github.com/advisories/GHSA-pxg6-pf52-xh8x
fix available via npm audit fix --force
Will install next-auth@4.24.7, which is a breaking change
node_modules/cookie
  @auth/core  <=0.35.3
  Depends on vulnerable versions of cookie
  node_modules/@auth/core
    next-auth  <=0.0.0-pr.11562.ed0fce23 || >=4.24.8
    Depends on vulnerable versions of @auth/core
    node_modules/next-auth
```

Problem is that I run AuthJS 5.0.0-beta.22 so downgrading to NextAuth 4.24 isn't really an option. Or am I missing something here?

I tried the --force just to see what would happen and it broke the app as expected.

Also: AuthJS 5 has been on beta.22 for what, a year at least now? But NextAuth is still getting regular releases. Is v5 still being developed or did I make a mistake by jumping onto the AuthJs train too early 🙂

Thanks for any tips!

**Comment by AOphagen** [comment]
I'd say I have the same problem with `next-auth`, without `@auth/core` in between. The fix for `cookie` is ready, so this should be an issue in `next-auth`, shouldn't it?
![Screenshot 2024-10-08 182948](https://github.com/user-attachments/assets/ab8a89bd-81cd-4970-9c9e-7e2ba7fc9839)

**Comment by chris-at-lemon** [comment]
I can see that `@auth/core` is at version 0.36.0 and that its `cookie `dependency is for 0.7.1, so in `@auth/core` it's been updated.

https://www.npmjs.com/package/@auth/core?activeTab=code

But I cannot find anywhere how to update AuthJs to use this new version of `@auth/core`

In fact I cannot find any documentation on how to update version of AuthJs.

Throw into the mix that NPM seem to think that AuthJs = Next-Auth and that on the GitHubh page both Next-Auth and AuthJs are linked to the same repo I've just been going around in circles for 2 days now.

My only guess is I could manually update dependencies in my package-lock file.

Honestly I'm a little surprised I seem to be the only one suffering from this. No answers on either Discord or here - on a security issue in an Auth library. Starting to doubt whether I made the right choice opting for this library.

**Comment by WikiRik** [comment, marked as answer]
`next-auth` version 5.0.0-beta.22 ships with 0.35.3 of `@auth/core` which is was the newest version at that time. It is now updated to 0.36.0 and `next-auth` just takes the version from the workspace; https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/package.json
Only thing that needs to be done imo is release a new beta of `next-auth` and the release pipeline will automatically update the version of `@auth/core` used.
@balazsorban44 can you run the pipeline for a new beta release of `next-auth`?

If people want to get rid of the `npm audit` message right now, with a bit of manual patching, you can use the `overrides` functionality in NPM; https://docs.npmjs.com/cli/v10/configuring-npm/package-json#overrides or `resolution` in Yarn; https://yarnpkg.com/cli/set/resolution to set `@auth/core` to 0.36.0 everywhere.

---

@chris-at-lemon this is not a security issue in `next-auth` itself but an underlying dependency and even a low vulnerability in there with a 0.045% chance of being exploited in the next month; https://github.com/advisories/GHSA-pxg6-pf52-xh8x
`next-auth` might not even use the vulnerable function. Read a bit more about false positives in `npm audit` here; https://overreacted.io/npm-audit-broken-by-design/ (Dan Abramov, the writer of the blog, is a core React team member)

Also; version 5.0.0-beta.22 was released two weeks ago; https://github.com/nextauthjs/next-auth/releases/tag/next-auth%405.0.0-beta.22 and not a year ago. Just a quick glance at the commits in this repository will tell you that there is much active development going on.

**Reply by chris-at-lemon** [reply]
Thank you for taking the time to reply @WikiRik!

on beta.22 - my bad, apologies, I thought I had .22 in my package file forever, but yu are right, it was .9, apologies.

On updating: even though this may be a non issue 99% of the time I cannot get a deploy with a vulnerability through the deployment pipeline in my company, it needs to be squeaky clean or automated tests raise red flags.

I will use overrides, thanks for confirming this will work.

**Reply by balazsorban44** [reply]
>  I cannot get a deploy with a vulnerability through the deployment pipeline

There is no vulnerability

**Reply by AOphagen** [reply]
Yes, understood, but how to explain that to a deployment pipeline? Will also try override.

P.S. & offtopic: 'squeaky clean' - made me laugh. Sorry, I envision a surgeon style scrubbing as preparation for something like using a door handle, An automation really doesn't read details and distinguish between usages of a dependency, it's sad.

**Comment by balazsorban44** [comment]
**TLDR;**

**To clarify there is _no_ vulnerability because of the `cookie` package in any of our packages.**

@WikiRik appreciate the assist/explanation!

`next-auth` will be updated in the next release, until then you can use the overrides option for `@auth/core` if need but to reiterate there is _no_ real vulnerability. 

@chris-at-lemon please _never_ report a potential vulnerability in a GitHub discussion or any public platform before consulting the maintainers per their security policy.

We have such document here https://authjs.dev/security

We never use user-provided values for name/domain/path of our cookies, it's all server-side.

That said, imagine if it _was_ an actual issue. I haven't seen this message before now, a day after you posted it, and only because someone tagged me. it's 2am here and am going to bed. If it was an actual issue, I would have to stay up the whole night to patch. had to done it before, it's unpleasant. 🙏 Besides, these kind of public reports often create unnecessary fear.

I have 100+ github mentions a day, unfortunately I cannot monitor all of them realistically/humanly, I wish I could.

No hard feelings, I'm just starting to emphasize the importance of reading the docs, especially in security sensitive cases like this.

I hope we learnt something today. 💚

**Reply by chris-at-lemon** [reply]
@balazsorban44 thanks for taking the time to reply, I appreciate your argument.

Just to reiterate myself: I did not intend to cause alarm or "report a vulnerability" I just asked how I can fix an audit warning. WikiRik kindly explained this.

The mix between next-auth and AuthJs surely confuses more people than just me. I just needed a little assistance, the GitHub discussion forum seemed the proper place to ask.

I appreciate this is a free solution and that you invest your time and knowledge. Thank you!

**Reply by balazsorban44** [reply]
Right, that's fair. Thanks for understanding!

Then I suggest dropping the words like "vulnerability" etc. from titles/description. because most people only akik through, instead of reading thoroughly. 🙏

For understanding Auth.js <> NextAuth.js, check out https://authjs.dev/contributors#history


I'm short, NextAuth.js is now just a subset, or rather a framework specific implementation of the core, Auth.js library, which is, fully agnostic to runtime or framework.

We call these integrations https://authjs.dev/guides/creating-a-framework-integration
